form/src/main/java/org/dhis2/form/ui/provider/inputfield/DateProvider.kt

```diff
@@ -42,14 +42,18 @@ fun ProvideInputDate(
         ValueType.TIME -> DateTimeActionType.TIME to TimeTransformation()
         else -> DateTimeActionType.DATE to DateTransformation()
     }
-    val textSelection = TextRange(if (fieldUiModel.value != null) fieldUiModel.value!!.length else 0)
+    val textSelection =
+        TextRange(if (fieldUiModel.value != null) fieldUiModel.value!!.length else 0)
     val yearIntRange = getYearRange(fieldUiModel)
     val selectableDates = getSelectableDates(fieldUiModel)
 
     var value by remember(fieldUiModel.value) {
         mutableStateOf(
             if (fieldUiModel.value != null) {
-                TextFieldValue(formatStoredDateToUI(fieldUiModel.value!!, fieldUiModel.valueType), textSelection)
+                TextFieldValue(
+                    formatStoredDateToUI(fieldUiModel.value!!, fieldUiModel.valueType),
```

Comment:
This could be a good opportunity to remove the !!

Reply:
won't do it for a hotfix